Modules/ArcGIS/DSCResources/ArcGIS_Portal_DisconnectedEnvironment/ArcGIS_Portal_DisconnectedEnvironment.psm1

```diff
@@ -0,0 +1,1042 @@
+                {
+                    ForEach ($geocodeService in $CurHelperServices.geocode)
+                    {
+                        if ($geocodeService.url -ne "https://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer")
+                        {
+                            $newGeocodeServices += $geocodeService
+                        }
+                    }
+                }
+            }
+
+            if ($HelperSrvcs.geocode.Services)
+            {
+                if (-not (Test-GeocodeUrls -CurGeocodeServices $CurHelperServices.geocode -GeocodeUrls $HelperSrvcs.geocode.Services))
+                {
+                    ForEach ($geocodeSrvc in $HelperSrvcs.geocode.Services.value)
+                    {
+                        $alreadyImplemented = $false
+                        ForEach ($newGeocodeService in $newGeocodeServices)
+                        {
+                            if ($newGeocodeService.url -eq $geocodeSrvc.url)
+                            {
+                                $alreadyImplemented = $true
+                                break
+                            }
+                        }
+                        if (-not $alreadyImplemented)
+                        {
+                            $geocodeInfos = Get-GeocodeInfos -GeocodeService $geocodeSrvc
+
+                            $newGeocodeServices += $geocodeInfos
+                        }
+                    }
+                }
+            }
+            $helperServiceParams.Add('geocodeService', (ConvertTo-Json $newGeocodeServices))
+        }
+
+        if($HelperSrvcs.geometry)
+        {
+            if ($HelperSrvcs.geometry.useHostedServer)
+            {
+                $HostedServerUrl = Get-HostedServerUrl -PortalUrl $PortalUrl -Token $($token.token)
+                if ($HostedServerUrl)
+                {
+                    if (-not (Test-GeometryStatus -ServerUrl $HostedServerUrl -Token $($token.token)))
+                    {
+                        Set-GeometryStatus -ServerUrl $HostedServerUrl -Token $($token.token)
+                    }
+
+                    if (-not (Test-GeometrySharing -ServerUrl $HostedServerUrl -PortalUrl $PortalUrl -Token $($token.token)))
+                    {
+                        Set-GeometrySharing -ServerUrl $HostedServerUrl -PortalUrl $PortalUrl -Token $($token.token)
+                    }
+                    
+                    if (-not ($CurHelperServices.geometry.url.StartsWith($HostedServerUrl)))
+                    {
+                        $serviceUrl = "$HostedServerUrl/rest/services/Utilities/Geometry/GeometryServer"
+                        $helperServiceParams.Add("geometryService",  '{"url": "' + $serviceUrl + '" }')
+                    }
+                } else {
+                    Write-Warning "No Hosted Server available. Geometry-Service is not working."
+                }
+            }
+            elseif ($HelperSrvcs.geometry.url)
```

Comment:
@spitzerr We will need another case for support for secure service which requires a credential. https://enterprise.arcgis.com/en/portal/latest/administer/linux/configure-services.htm#ESRI_SECTION2_CD97431BAC104D9A80DEA0D55F4420D9